fairscale/experimental/wgit/signal_sparsity.py

```diff
@@ -63,17 +82,41 @@ def _validate_conf(self) -> None:
 
         this should assert fail if checking fails.
         """
-        pass
+        # assert that both top_k_elements and top_k_percent aren't set for sst and dst
+        assert (
+            self._sst_top_k_element is None or self._sst_top_k_percent is None
+        ), "Both top-k element and top-k percent has been provided as argument"
+        assert (
+            self._dst_top_k_element is None or self._dst_top_k_percent is None
+        ), "Both top-k element and top-k percent has been provided as argument"
 
     def dense_to_sst(self, dense: Tensor) -> Tensor:
         """Get SST from a tensor
 
         Dense -> fft -> top-k -> results.
 
+        For sparsification (or getting the topk) of the tensor in the transformed domain, the sparse
+        mask needs to be generated from the absolute values of the transformed tensors. However, using
+        the mask, we mask out the top_k values from the transformed tensor `w_fft` itself instead.
+        Note: the transformed tensor `w_fft` is a tensor of complex values, whereas `w_fft_abs` is a
+        tensor with real values from which we generate the sparse mask.
+
         Returns:
             Same shaped tensor, still in dense format but in frequency domain and has zeros.
         """
-        pass
+        if self._sst_top_k_element is not None:
+            assert self._sst_top_k_dim is None or (
+                self._sst_top_k_element <= dense.shape[self._sst_top_k_dim]
+            ), f"top_k_element along dim {self._sst_top_k_dim} out of range for tensor of shape {dense.shape}."
+
+        w_fft = self._transform(dense)
+        w_fft_abs = torch.abs(w_fft)  # get absolute FFT values
```

Comment:
you can use w_fft.abs()? Also, for a complex number, what's the abs() doing to it? The comment just repeats what the code is doing, but the comment does explain why abs() is OK here.

Reply:
the abs is calculating the magnitude of the complex values, similar to the magnitude of 2D vectors. I am adding an explanatory comment on this for clarification. Also thanks for suggesting the `w_fft.abs()` change.